src/main/java/forum/service/ThreadServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package forum.service;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import forum.dao.ThreadDAO;
+import forum.entity.Thread;
+
+@Service
+public class ThreadServiceImpl implements ThreadService {
+	
+	@Autowired
+	private ThreadDAO threadDAO;
+
+	@Override
+	@Transactional
+	public void saveThread(Thread thread) {
+		threadDAO.saveThread(thread);
```

Comment:
More more logic here from the controller

Reply:
Moved the logic to the ThreadService